lib/at_cmd/at_cmd.c

```diff
@@ -111,23 +134,114 @@ static int get_return_code(char *buf, struct return_state_object *ret)
 
 static void callback_worker(struct k_work *item)
 {
-	struct callback_work_item *data =
-		CONTAINER_OF(item, struct callback_work_item, work);
+	ARG_UNUSED(item);
+	struct resp_item resp;
 
-	if (data != NULL) {
-		data->callback(data->data);
+	while (k_msgq_get(&responses, &resp, K_NO_WAIT) == 0) {
+		if (resp.callback != NULL ||
+		    resp.buf != NULL) {
+			resp.callback(resp.buf);
+		}
+		if (resp.flags & AT_RESP_ALLOC) {
```

Comment:
Again, I wonder if we need this flag. Isn't `callback_worker` processing only asynchronous commands and notifications, that will always have this buffer dynamically allocated (with current implementation)?

Reply:
Removed

Reply:
No longer relevant